### PR TITLE
[JENKINS-57023] Switch dep from `workflow-cps-global-lib` to `pipeline-groovy-lib`

### PR DESCRIPTION
**Pull request by jglick**
See https://github.com/jenkinsci/workflow-cps-global-lib-plugin/pull/172.

### CODE REVIEW

File: pom.xml
Comment (jglick) [inline pom.xml:64]: Minimum version for deps.

File: pom.xml
Comment (jglick) [inline pom.xml:93]: Same as in BOM.

**Review by MarkEWaite — APPROVED** [review]

File: pom.xml
Comment (MarkEWaite) [inline pom.xml:78]: Since we're upgrading the minimum required Jenkins version and the matching plugin bom version, shouldn't we also upgrade the version of the plugin bom?
```suggestion
        <artifactId>bom-2.303.x</artifactId>
        <version>1382.v7d694476f340</version>
```

**Review by MarkEWaite — APPROVED** [review]
Optional comment proposing upgrade of the plugin bom release along with the increase of minimum Jenkins version

File: pom.xml
Comment (jglick) [reply pom.xml:78]: You could. It was not necessary for this purpose so I left it at the maintainer’s discretion to do that now or in some independent PR.

**Review by car-roll — APPROVED** [review]

File: pom.xml
Comment (jglick) [reply pom.xml:64]: https://github.com/jenkinsci/workflow-cps-global-lib-plugin/pull/119#issuecomment-1136531975 however

**Comment by jetersen** [comment]
This is still waiting on a release 👍

**Comment by MarkEWaite** [comment]
> This is still waiting on a release 👍

I hope to release a new version in the next 3-4 weeks.

**Pull request merged**
